extensions/sql-database-projects/src/test/dialogs/addSqlBindingQuickpick.test.ts

```diff
@@ -77,5 +80,49 @@ describe('Add SQL Binding quick pick', () => {
 		should(spy.calledOnce).be.true('showErrorMessage should have been called exactly once');
 		should(spy.calledWith(errormsg)).be.true(`showErrorMessage not called with expected message '${errormsg}' Actual '${spy.getCall(0).args[0]}'`);
 	});
-});
 
+	it('Should show error connection profile does not connect', async function (): Promise<void> {
+		sinon.stub(utils, 'getAzureFunctionService').resolves(testContext.azureFunctionService.object);
+		sinon.stub(utils, 'getVscodeMssqlApi').resolves(testContext.vscodeMssqlIExtension.object);
+		let connectionCreds = createTestCredentials();
+
+		sinon.stub(azureFunctionUtils, 'getAFProjectContainingFile').resolves(vscode.Uri.file('testUri'));
+		testContext.azureFunctionService.setup(x => x.getAzureFunctions(TypeMoq.It.isAny())).returns(async () => {
+			return Promise.resolve({
+				success: true,
+				errorMessage: '',
+				azureFunctions: ['af1']
+			});
+		});
+
+		// Mocks connect call to mssql
+		let error = new Error('Connection Request Failed');
+		testContext.vscodeMssqlIExtension.setup(x => x.connect(TypeMoq.It.isAny(), undefined)).throws(error);
+
+		// Mocks promptForConnection
+		testContext.vscodeMssqlIExtension.setup(x => x.promptForConnection(true)).returns(() => Promise.resolve(connectionCreds));
+		let quickpickStub = sinon.stub(vscode.window, 'showQuickPick')
+
+		// select Azure function
+		quickpickStub.onFirstCall().resolves({ label: 'af1' });
+		// select input or output binding
+		quickpickStub.onSecondCall().resolves({ label: constants.input });
+
+		// give object name
+		let inputBoxStub = sinon.stub(vscode.window, 'showInputBox').onFirstCall().resolves('dbo.table1');
+
+		// select connection profile
+		quickpickStub.onThirdCall().resolves({ label: constants.createNewLocalAppSettingWithIcon });
+
+		// give connection string setting name
+		inputBoxStub.onSecondCall().resolves('SqlConnectionString');
+
+		// select connection profile method
+		quickpickStub.onCall(3).resolves({ label: constants.connectionProfile });
+
+		await launchAddSqlBindingQuickpick(vscode.Uri.file('testUri'), packageHelper);
+
+		// should go back to the select connection string methods
+		should(quickpickStub.callCount === 5);
```

Comment:
To truly verify this you should be verifying that not only was it called but that it was called with the expected titles. Since otherwise it could show any quickpick (such as the next one in the flow) and it would still pass. Sinon has a `.withArgs` function that you could use to create separate spies for each type of quickpick we display (based on title) and then verify the count for each of those.

Reply:
since I only need to verify the one call. I checked that.